tests/dragonfly/conftest.py

```diff
@@ -225,3 +226,40 @@ def port_picker():
 @pytest.fixture(scope="class")
 def memcached_connection(df_server: DflyInstance):
     return pymemcache.Client(f"localhost:{df_server.mc_port}")
+
+
+@pytest.fixture(scope="function")
+def tls_server_key_file_name():
+    return "df-key.pem"
+
+@pytest.fixture(scope="function")
+def tls_server_cert_file_name():
+    return "df-cert.pem"
+
+@pytest.fixture(scope="function")
+def gen_tls_cert(df_factory: DflyInstanceFactory, tls_server_key_file_name, tls_server_cert_file_name):
+    dfly_path = df_factory.dfly_path
+    # We first need to generate the tls certificates to be used by the server
+
+    # Step 1
+    # Generate CA (certificate authority) key and self-signed certificate
+    # In production, CA should be generated by a third party authority
+    # Expires in one day and is not encrtypted (-nodes)
+    # X.509 format for the key
+    ca_key = dfly_path + "ca-key.pem"
+    ca_cert = dfly_path + "ca-cert.pem"
+    step1 = rf'openssl req -x509 -newkey rsa:4096 -days 1 -nodes -keyout {ca_key} -out {ca_cert} -subj "/C=GR/ST=SKG/L=Thessaloniki/O=KK/OU=AcmeStudios/CN=Gr/emailAddress=acme@gmail.com"'
+    subprocess.run(step1, shell=True)
+
+    # Step 2
+    # Generate Dragonfly's private key and certificate signing request (CSR)
+    tls_server_key = dfly_path + tls_server_key_file_name
+    tls_server_req = dfly_path + "df-req.pem"
+    step2 = rf'openssl req -newkey rsa:4096 -nodes -keyout {tls_server_key} -out {tls_server_req} -subj "/C=GR/ST=SKG/L=Thessaloniki/O=KK/OU=Comp/CN=Gr/emailAddress=does_not_exist@gmail.com"'
+    subprocess.run(step2, shell=True)
+
+    # Step 3
+    # Use CA's private key to sign dragonfly's CSR and get back the signed certificate
+    tls_server_cert = dfly_path + tls_server_cert_file_name
+    step3 = fr'openssl x509 -req -in {tls_server_req} -days 1 -CA {ca_cert} -CAkey {ca_key} -CAcreateserial -out {tls_server_cert}'
+    subprocess.run(step3, shell=True)
```

Comment:
gen_tls_cert can return the path of key and certificate file names and instead of defining tls_server_cert_file_name and tls_server_key_file_name set the filenames to constant var